### PR TITLE
#28 - CRUD for schools backend

### LINKED ISSUE #28: CRUD for schools - Backend
We need to implement CRUD functionality for schools, ensuring each school has a name and address. Additionally, we require an option to fetch data from the RSPO API.

### PR DESCRIPTION
**Pull request by AmonDeShir**
Should close #28.

### CODE REVIEW

File: app/DTO/SchoolDTO.php [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+class SchoolDTO
+{
+    public function __construct(
+        public string $name,
+        public string $city,
+        public string $street,
+        public string $building_number,
+        public string $apartment_number,
+        public string $zip_code,
```
Comment (kamilpiech97) [inline app/DTO/SchoolDTO.php:15]:
```suggestion
        public string $buildingNumber,
        public string $apartmentNumber,
        public string $zipCode,
```

File: app/Helpers/SchoolTypeHelper.php [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+namespace App\Helpers;
+
+class SchoolTypeHelper
+{
+    public const int LICEUM_OGOLNOKSZTALCACE = 14;
+    public const int TECHNIKUM = 16;
+    public const int LICEUM_UZUPELNIAJACE = 17;
+    public const int BEDNARSKA_SZKOLA_REALNA = 90;
+    public const int SZKOLA_BRANZOWA_DRUGIEGO_STOPNIA = 94;
```
Comment (kamilpiech97) [inline app/Helpers/SchoolTypeHelper.php:13]: Why that numbers?

File: config/saloon.php [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+    | By default, this package will create any classes within
+    | `/app/Http/Integrations` directory. If you're using
+    | a different design approach, then your classes
+    | may be in a different place. You can change
+    | that location so that the saloon:list
+    | command will still find your files
+    |
+    */
+
+    "integrations_path" => base_path("App/Http/Integrations"),
```
Comment (kamilpiech97) [inline config/saloon.php:36]:
```suggestion
    "default_sender" => GuzzleSender::class,
    "integrations_path" => base_path("App/Http/Integrations"),
```

**Review by kamilpiech97 — CHANGES_REQUESTED** [review]

File: app/Http/Controllers/SchoolsController.php [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+        return Inertia::render("SchoolsPanel", ["schools" => SchoolResource::collection(School::all())]);
+    }
+
+    public function store(SchoolRequest $request): RedirectResponse
+    {
+        School::query()->create($request->validated());
+
+        return redirect()
+            ->back()
+            ->with("success", "School added successfully");
```
Comment (AleksandraKozubal) [inline app/Http/Controllers/SchoolsController.php:35]: Is the administration panel in Polish? If so, translate these messages

Comment (AmonDeShir) [reply app/Http/Controllers/SchoolsController.php:35]: done

File: app/DTO/SchoolDTO.php (hunk 1 above)
Comment (AmonDeShir) [reply app/DTO/SchoolDTO.php]: done

File: app/Http/Integrations/RSPOConnector/RSPOConnector.php [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+    public function paginate(Request $request): Paginator
+    {
+        return new RSPOPaginator($this, $request);
+    }
+
+    protected function defaultHeaders(): array
+    {
+        return [
+            "Accept" => "application/ld+json",
+        ];
```
Comment (kamilpiech97) [inline app/Http/Integrations/RSPOConnector/RSPOConnector.php:31]:
```suggestion
        return ["Accept" => "application/ld+json"];
```

File: app/Http/Integrations/RSPOConnector/RSPOConnector.php [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+    }
+
+    protected function defaultConfig(): array
+    {
+        return [];
+    }
```
Comment (kamilpiech97) [inline app/Http/Integrations/RSPOConnector/RSPOConnector.php:37]:
```suggestion
```
Is it needed?

File: app/Http/Controllers/SchoolsController.php [hunk 7]
```diff
@@ -0,0 +1,93 @@
...
+    /**
+     * @throws Throwable
+     */
+    public function fetch(): JsonResponse
+    {
+        if ($this->isFetching()) {
+            return response()->json(["message" => "Pobieranie w toku, proszę czekać"], Status::HTTP_CONFLICT);
+        }
+
+        $jobs = [new FetchSchoolsJob(VoivodeshipsHelper::LOWER_SILESIA)];
```
Comment (kamilpiech97) [inline app/Http/Controllers/SchoolsController.php:62]: Lets load that from config

File: app/Http/Integrations/RSPOConnector/Requests/GetSchoolsRequest.php [hunk 8]
```diff
@@ -0,0 +1,52 @@
...
+use Saloon\Http\Response;
+use Saloon\PaginationPlugin\Contracts\Paginatable;
+
+class GetSchoolsRequest extends Request implements Paginatable
+{
+    protected Method $method = Method::GET;
+
+    public function __construct(
+        protected string $voivodeship,
+        protected int $school_type,
```
Comment (kamilpiech97) [inline app/Http/Integrations/RSPOConnector/Requests/GetSchoolsRequest.php:20]:
```suggestion
        protected int $schoolType,
```

File: app/Http/Integrations/RSPOConnector/Requests/GetSchoolsRequest.php [hunk 9]
```diff
@@ -0,0 +1,52 @@
...
+    ) {}
+
+    public function defaultQuery(): array
+    {
+        return [
+            "zlikwidowana" => false,
+            "typ_podmiotu_id" => $this->school_type,
```
Comment (kamilpiech97) [inline app/Http/Integrations/RSPOConnector/Requests/GetSchoolsRequest.php:27]:
```suggestion
            "typ_podmiotu_id" => $this->schoolType,
```

**Review by kamilpiech97 — CHANGES_REQUESTED** [review]

File: app/Http/Integrations/RSPOConnector/Requests/GetSchoolsRequest.php (hunk 9 above)
Comment (AmonDeShir) [reply app/Http/Integrations/RSPOConnector/Requests/GetSchoolsRequest.php:27]: done

File: app/Http/Controllers/SchoolsController.php (hunk 7 above)
Comment (AmonDeShir) [reply app/Http/Controllers/SchoolsController.php]: done

File: app/Helpers/VoivodeshipsHelper.php
Comment (kamilpiech97) [inline app/Helpers/VoivodeshipsHelper.php:1]: Why did u remove that? U can use Helper or Enum (btter option) in config file.

File: config/schools.php [hunk 10]
```diff
@@ -0,0 +1,9 @@
+<?php
+
+declare(strict_types=1);
+
+return [
+    "voivodeships" => [
+        "DOLNOŚLĄSKIE",
```
Comment (kamilpiech97) [inline config/schools.php:7]:
```suggestion
        "DOLNOŚLĄSKIE",
```
Create for voivodeships Enum and load here enabled values.

Comment (AmonDeShir) [reply config/schools.php]: done

File: app/Helpers/VoivodeshipsHelper.php
Comment (AmonDeShir) [reply app/Helpers/VoivodeshipsHelper.php:1]: done

**Review by kamilpiech97 — APPROVED** [review]

**Review by mlencki — APPROVED** [review]

**Pull request merged**